src/BSON/Timestamp.c

```diff
@@ -226,6 +232,102 @@ PHP_METHOD(Timestamp, __wakeup)
 }
 /* }}} */
 
+/* {{{ proto string Timestamp::serialize()
+*/
+PHP_METHOD(Timestamp, serialize)
+{
+	php_phongo_timestamp_t   *intern;
+#if PHP_VERSION_ID >= 70000
+	zval                      retval;
+#else
+	zval                     *retval;
+#endif
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+	char                      s_increment[24];
+	char                      s_timestamp[24];
```

Comment:
If `PRIu32` is used for these unsigned, 32-bit values, the largest string representation would be "4294967295" (11 characters including the null byte). I think 12 is sufficient here.

24 was fine for 64-bit values, which could be 21 characters including the null byte (signed or unsigned).